src/App.tsx

```diff
@@ -15,47 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function getVisibleGoods(
+  goods: string[],
+  sortField: SortType,
+  toReverse: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortField) {
+    visibleGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (toReverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReversed, setReversedField] = useState(false);
+
+  const visibleGoods = getVisibleGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+
+  const handleResetClick = () => {
+    setSortField(SortType.DEFAULT);
+    setReversedField(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortType.ALPHABET })}
+          onClick={() => setSortField(SortType.ALPHABET)}
```

Comment:
all the functions inside onClick should be moved to separate functions like `handleResetClick`